internal/relay/relayer.go

```diff
@@ -92,28 +87,10 @@ func (r *Relayer) Run(ctx context.Context, tasks <-chan neutrontypes.RegisteredQ
 			}
 		case <-ctx.Done():
 			r.logger.Info("Context cancelled, shutting down relayer...")
-			return r.stop()
```

Comment:
incorrect since we need to return here (cause we're inside the infinite loop)?

Reply:
yeah, a `return` here is certainly missing

Reply:
true